### PR TITLE
MULE-15532: Make the WSC use the Soap Core library

### PR DESCRIPTION
**Pull request by juandesi**

### CODE REVIEW

File: src/main/java/org/mule/extension/ws/internal/connection/SoapClientConnectionProvider.java [hunk 1]
```diff
@@ -115,17 +108,26 @@
...
    */
   @Override
-  public SoapClientWrapper connect() throws ConnectionException {
-    return new SoapClientWrapper(soapService.getClientFactory().create(getConfiguration()), customTransportConfiguration);
+  public WscSoapClient connect() throws ConnectionException {
+    try {
+      if (customTransportConfiguration == null) {
+        customTransportConfiguration = new DefaultHttpTransportConfiguration(client);
+      }
+      return new WscSoapClient(info.getWsdlLocation(),
```
Comment (alegmarra) [inline src/main/java/org/mule/extension/ws/internal/connection/SoapClientConnectionProvider.java]: when is the wsdl parsed?

File: src/main/java/org/mule/extension/ws/internal/error/WscError.java [hunk 2]
```diff
@@ -4,16 +4,16 @@
...
  *
  * @since 1.0
  */
-public class WscException extends RuntimeException {
+public enum WscError implements ErrorTypeDefinition<WscError> {
 
-  public WscException(String message, Throwable cause) {
-    super(message, cause);
-  }
+  INVALID_WSDL, ENCODING, BAD_REQUEST, BAD_RESPONSE, CANNOT_DISPATCH, SOAP_FAULT
```
Comment (alegmarra) [inline src/main/java/org/mule/extension/ws/internal/error/WscError.java]: docs

File: src/main/java/org/mule/extension/ws/internal/error/WscExceptionEnricher.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  @Override
+  public Exception enrichException(Exception e) {
+    if (e instanceof SoapFaultException) {
+      return new SoapFaultMessageAwareException(((SoapFaultException) e));
+    }
+    WscError error = ERRORS_MAPPING.get(e.getClass());
+    if (error != null) {
+      return new ModuleException(error, e);
+    }
+    return new Exception("Unexpected error while consuming web service: " + e.getMessage(), e);
```
Comment (alegmarra) [inline src/main/java/org/mule/extension/ws/internal/error/WscExceptionEnricher.java]: `WscException`

File: src/main/java/org/mule/extension/ws/internal/error/ConsumeErrorTypeProvider.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+ * @since 1.0
+ */
+public class ConsumeErrorTypeProvider implements ErrorTypeProvider {
+
+  /**
+   * @return all the error types that can be thrown by the {@link ConsumeOperation}.
+   */
+  @Override
+  public Set<ErrorTypeDefinition> getErrorTypes() {
+    return ImmutableSet.of(BAD_REQUEST, BAD_RESPONSE, ENCODING, INVALID_WSDL, SOAP_FAULT, CANNOT_DISPATCH);
```
Comment (estebanwasinger) [inline src/main/java/org/mule/extension/ws/internal/error/ConsumeErrorTypeProvider.java]: `TIMEOUT` Doesn't exist anymore and breaks backwards

File: src/main/java/org/mule/extension/ws/internal/metadata/ConsumeOutputResolver.java [hunk 5]
```diff
@@ -7,35 +7,69 @@
...
+
+  @Override
+  public String getCategoryName() {
+    return WebServiceConsumer.NAME;
+  }
 
   @Override
   public String getResolverName() {
-    return "ConsumeOutputResolver";
+    return WebServiceConsumer.NAME + " Output Resolver";
```
Comment (alegmarra) [inline src/main/java/org/mule/extension/ws/internal/metadata/ConsumeOutputResolver.java]: either all with space after the `+` or none

File: src/main/java/org/mule/extension/ws/internal/connection/SoapClientConnectionProvider.java (hunk 1 above)
Comment (juandesi) [reply src/main/java/org/mule/extension/ws/internal/connection/SoapClientConnectionProvider.java]: the engine parses it

File: src/main/java/org/mule/extension/ws/api/transport/CustomHttpTransportConfiguration.java [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+ * Provides the capability to send soap messages using a custom HTTP Requester configuration.
+ *
+ * @since 1.1
+ */
+@Alias("http-transport-configuration")
+public class CustomHttpTransportConfiguration implements CustomTransportConfiguration {
+
+  @ConfigReference(namespace = "HTTP", name = "REQUEST_CONFIG")
+  @Parameter
+  private String requesterConfig;
```
[the remark this answers is not in the document]
Comment (juandesi) [reply src/main/java/org/mule/extension/ws/api/transport/CustomHttpTransportConfiguration.java]: sadly this accepts expressions :/

**Pull request merged**